invenio_vocabularies/datastreams/readers.py

```diff
@@ -258,19 +258,38 @@ def __init__(
 
     def _iter(self, scythe, *args, **kwargs):
         """Read and parse an OAIPMH stream to dict."""
-        scythe.class_mapping["ListRecords"] = self.OAIRecord
-        try:
-            records = scythe.list_records(
-                from_=self._from,
-                until=self._until,
-                metadata_prefix=self._metadata_prefix,
-                set_=self._set,
-                ignore_deleted=True,
-            )
-            for record in records:
-                yield {"record": record}
-        except NoRecordsMatch:
-            raise ReaderError(f"No records found in OAI-PMH request.")
+        if self._verb == "ListRecords":
+            scythe.class_mapping["ListRecords"] = self.OAIRecord
+            try:
+                records = scythe.list_records(
+                    from_=self._from,
+                    until=self._until,
+                    metadata_prefix=self._metadata_prefix,
+                    set_=self._set,
+                    ignore_deleted=True,
+                )
+                for record in records:
+                    yield {"record": record}
+            except NoRecordsMatch:
+                raise ReaderError(f"No records found in OAI-PMH request.")
+        else:
```

Comment:
Are there other verbs folks might use? Should there be an error that they aren't implemented?

Reply:
I see your point, because the design follows the verbs that are defined by OAI-PMH. I chose them to offer familiar terms to the admins and wasn't aware of the drawback. But the usage is safe in the sense you can put in every verb you like and the harvester does it's work (perhaps in a way the suer doesn't expect).